flink/v1.14/build.gradle

```diff
@@ -119,7 +119,6 @@ project(":iceberg-flink:iceberg-flink-runtime-${flinkMajorVersion}") {
 
   configurations {
     implementation {
-      exclude group: 'org.apache.flink'
```

Comment:
Nice catch. 

Can we add a smoke test to Flink runtime jar to catch these kinds of issues?
 Spark is already having one here for its runtime jar. https://github.com/apache/iceberg/blob/master/spark/v3.3/spark-runtime/src/integration/java/org/apache/iceberg/spark/SmokeTest.java

Reply:
good point, I added a smoke test that was failing/hanging without this fix and works with this fix